### PR TITLE
[rocSPARSE] Interface additional sparse routines

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: src/sparse/generic.jl [hunk 1]
```diff
@@ -2,13 +2,123 @@
...
+    desc_sparse = ROCSparseMatrixDescriptor(A, index)
+    desc_dense = ROCDenseMatrixDescriptor(B)
+
+    function bufferSize()
+        out = Ref{Csize_t}()
+        rocsparse_sparse_to_dense(handle(), desc_sparse, desc_dense, out, C_NULL)
+        return out[]
+    end
+    with_workspace(bufferSize) do buffer
+        buffer_size = sizeof(buffer)
```
Comment (amontoison) [inline src/sparse/generic.jl:18]:
```suggestion
        buffer_size[] = sizeof(buffer)
```

File: src/sparse/generic.jl [hunk 2]
```diff
@@ -2,13 +2,123 @@
...
+    B = ROCMatrix{T}(undef, m, n)
+    desc_sparse = ROCSparseMatrixDescriptor(A, index)
+    desc_dense = ROCDenseMatrixDescriptor(B)
+
+    function bufferSize()
+        out = Ref{Csize_t}()
+        rocsparse_sparse_to_dense(handle(), desc_sparse, desc_dense, out, C_NULL)
+        return out[]
+    end
+    with_workspace(bufferSize) do buffer
```
Comment (amontoison) [inline src/sparse/generic.jl:17]:
```suggestion
    buffer_size = Ref{Csize_t}()
    with_workspace(bufferSize) do buffer
```

File: src/sparse/generic.jl [hunk 3]
```diff
@@ -2,13 +2,123 @@
...
+    end
+    desc_dense = ROCDenseMatrixDescriptor(A)
+
+    function bufferSize()
+        out = Ref{Csize_t}()
+        rocsparse_dense_to_sparse(handle(), desc_dense, desc_sparse, algo, out, C_NULL)
+        return out[]
+    end
+    with_workspace(bufferSize) do buffer
+        buffer_size = sizeof(buffer)
```
Comment (amontoison) [inline src/sparse/generic.jl:47]:
```suggestion
        buffer_size[] = sizeof(buffer)
```

File: src/sparse/generic.jl [hunk 4]
```diff
@@ -2,13 +2,123 @@
...
+        error("Format :$fmt not available, use :csc, :csr or :coo.")
+    end
+    desc_dense = ROCDenseMatrixDescriptor(A)
+
+    function bufferSize()
+        out = Ref{Csize_t}()
+        rocsparse_dense_to_sparse(handle(), desc_dense, desc_sparse, algo, out, C_NULL)
+        return out[]
+    end
+    with_workspace(bufferSize) do buffer
```
Comment (amontoison) [inline src/sparse/generic.jl:46]:
```suggestion
    buffer_size = Ref{Csize_t}()
    with_workspace(bufferSize) do buffer
```

**Comment by pxl-th** [comment]
You can rebase this branch on master to get rid of 1.11.2 errors (ref: #705).

**Pull request merged**